tests/test_predefined_dynamical_decoupling.py

```diff
@@ -670,18 +674,19 @@ def test_attribute_values():
         )
 
 
-def _pulses_produce_identity(sequence):
+def _pulses_produce_identity(sequence, extra_rotation=None):
```

Comment:
I'd tentatively suggest changing this so that it always checks for I or Z, since I think that's really the property that we're always trying to check with this function.

If we *do* want to be more prescriptive (by checking for a specific operation), I'd suggest changing this either to accept the full expected operator all the time (e.g. `def _check_pulses(sequence, expected_operator)` or something), or to return the operator produced by the pulses so that the caller can check it themselves (e.g. `def _get_operator(sequence)`).

Reply:
SG. Remove `extra_rotation` and check explicitly Z rotation in return.